spec/spec_helper.rb

```diff
@@ -42,20 +42,20 @@ def set_node_version(version)
   end
 end
 
+def resolve_binary_path
+  (/darwin/ =~ RUBY_PLATFORM) != nil ? './vendor/resolve-version-darwin' : './vendor/resolve-version-linux'
+end
+
 def resolve_node_version(requirements, options = {})
-  # use nodebin to get latest node versions
   requirements.map do |requirement|
-    retry_limit = options[:retry_limit] || 50
-    body = Excon.get("https://nodebin.herokai.com/v1/node/linux-x64/latest?range=#{requirement}", :idempotent => true, :expects => 200, :retry_limit => retry_limit).body
-    JSON.parse(body)['number']
+    result = `#{resolve_binary_path} node #{requirement}`
+    result.split(' ')[0]
   end
 end
 
 def resolve_all_supported_node_versions(options = {})
-  retry_limit = options[:retry_limit] || 50 
-  body = Excon.get("https://nodebin.herokai.com/v1/node/linux-x64/", :idempotent => true, :expects => 200, :retry_limit => retry_limit).body
-  list = JSON.parse(body).map { |n| n['number'] }
-
+  result = `#{resolve_binary_path} list node`
+  list = result.lines().map { |line| line.split(' ')[0] }
```

Comment:
What you've got is fine. Ideomatic ruby would probably be 

```
   list = result.each_line.map { |line| line.split(' ').first } 
```

The `each_line` returns an enumerator rather than an array, which is a little cheaper than allocating an array. The `first` method is typically preferred over directly indexing into an array output, though really there's nothing wrong with either.